compiler/rustc_trait_selection/src/traits/project.rs

```diff
@@ -621,8 +623,13 @@ struct Progress<'tcx> {
 }
 
 impl<'tcx> Progress<'tcx> {
-    fn error(tcx: TyCtxt<'tcx>, guar: ErrorGuaranteed) -> Self {
-        Progress { term: Ty::new_error(tcx, guar).into(), obligations: PredicateObligations::new() }
+    fn error(tcx: TyCtxt<'tcx>, alias_term: ty::AliasTerm<'tcx>, guar: ErrorGuaranteed) -> Self {
```

Comment:
```suggestion
    fn error_of_kind(tcx: TyCtxt<'tcx>, term_kind: ty::AliasTerm<'tcx>, guar: ErrorGuaranteed) -> Self {
```
or `error_for_term` :thinking: otherwise the `alias_term` arg is a bit confusing